test/integration/framework/garden_operation.go

```diff
@@ -63,6 +63,25 @@ const (
 
 // NewGardenTestOperation initializes a new test operation from created shoot Objects that can be used to issue commands against seeds and shoots
 func NewGardenTestOperation(ctx context.Context, k8sGardenClient kubernetes.Interface, logger logrus.FieldLogger, shoot *v1beta1.Shoot) (*GardenerTestOperation, error) {
+	operation := &GardenerTestOperation{
+		Logger:       logger,
+		GardenClient: k8sGardenClient,
+	}
+	if shoot != nil {
+		if err := operation.AddShoot(ctx, shoot); err != nil {
+			return nil, errors.Wrapf(err, "could not add shoot to operation")
+		}
+	}
+
+	return operation, nil
+}
+
+// AddShoot sets the shoot and its seed for the GardenerOperation.
+func (o *GardenerTestOperation) AddShoot(ctx context.Context, shoot *v1beta1.Shoot) error {
```

Comment:
Wouldn't be better to have:
```go
func NewGardenTestOperation(ctx context.Context, k8sGardenClient kubernetes.Interface, logger logrus.FieldLogger) (*GardenerTestOperation, error)

func NewGardenTestOperationWithShoot(ctx context.Context, k8sGardenClient kubernetes.Interface, logger logrus.FieldLogger, shoot *v1beta1.Shoot) (*GardenerTestOperation, error)
```

The idea is to prevent passing `nil` when you don't need the optional arg.